homeassistant/helpers/entity_platform.py

```diff
@@ -464,6 +465,14 @@ def add_entities(
             self._async_unsub_polling()
             self._async_unsub_polling = None
 
+    @callback
+    def async_destroy(self) -> None:
```

Comment:
I'm marking this as WIP because this method indicates that some abstractions are leaking. Let me think about it a bit more.